### DISCUSSION TITLE
Feature Request: table context menu

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI-react#646] [Ideas]
It would be nice to have a native implementation of the dropdown menu inside the table tied to a right click handler. For tables with selection, right-click, ctrl+right-click can behave the same as left click, plus open the context menu. The menu opens positionally where clicked. It would need the consumer to give it a function which can take the selected rows and provide the contents of the menu.

**Comment by mayank99** [comment]
Are you facing any difficulties building your own context menu? I don't think we would want to add any specific implementations like the ones you mention.

**Reply by HaveSpacesuit** [reply]
Currently, I'm putting a dropdown menu on each row, which works, but isn't the best. There was also a CSS issue I battled today; we hide the button if the row isn't hovered or the menu isn't open, but this was causing a lot of jitteriness when combined with virtualization.
